code/modules/vehicles/mecha/_mecha.dm

```diff
@@ -561,14 +574,25 @@
 ///Called when a driver clicks somewhere. Handles everything like equipment, punches, etc.
 /obj/vehicle/sealed/mecha/proc/on_mouseclick(mob/user, atom/target, list/modifiers)
 	SIGNAL_HANDLER
+	if(LAZYACCESS(modifiers, MIDDLE_CLICK))
+		weapons_safety = !weapons_safety
+		SEND_SOUND(user, sound('sound/machines/beep.ogg', volume=25))
+		balloon_alert(user, "equipment [weapons_safety ? "safe" : "ready"]")
+		set_mouse_pointer()
+		return COMSIG_MOB_CANCEL_CLICKON
+	if(isAI(user)) //Middle clicks are already handled for AIs via above
+		if(weapons_safety)
+			return //Use AI functions
+		else
+			. = COMSIG_MOB_CANCEL_CLICKON //Use Mech functions
+	if(weapons_safety)
+		return
```

Comment:
```suggestion
	if(weapons_safety)
		return
	if(isAI(user))
		// For AIs: If safeties are off, use mech functions. If safeties are on, use AI functions. 
		. = COMSIG_MOB_CANCEL_CLICKON 
```

Reply:
For both AIs and non-AIs, we early return if `weapons_safety` is `TRUE`. 
The only difference is ais set `COMSIG_MOB_CANCEL_CLICKON` if `weapons_safety` is `FALSE`.